MonstieWash/Assets/Scripts/Consumables/ConsumablesManager.cs

```diff
@@ -137,17 +124,32 @@ public void toggleBag()
     public void RefreshUI()
     {
         //Generate new Ui objects
-        foreach (consumableData consumeStruct in consumableList)
+        foreach (Inventory.InventoryEntry itemData in m_inventory.StoredItemsList)
         {
-            //Check if consumable already has Ui object and skip
-            var matched = false; 
-            foreach (UIConsumable uiConsumable in activeUiElements)
+            if (itemData.ItemData.ContainsTag(Inventory.ItemTags.Consumable))
             {
-                if (consumeStruct.Consumable.ConsumableName == uiConsumable.consumable.ConsumableName) matched = true;
-            }
-            if (matched) continue;
+                //Check if consumable already has Ui object and skip
+                var matched = false;
+                int consumableToRemoveIndex = -1;
 
-            CreateUiElement(consumeStruct);
+                foreach (UIConsumable uiConsumable in activeUiElements)
+                {
+                    if (itemData.ItemData.ItemName == uiConsumable.consumable.ConsumableName) matched = true;
```

Comment:
String equality is kinda unreliable when done like this; the proper way is by using `String.Equals()` method.

```suggestion
                    if (itemData.ItemData.ItemName.Equals(uiConsumable.consumable.ConsumableName)) matched = true;
```